### PR TITLE
Nette3

### PR DESCRIPTION
**Pull request by solcik**
This change was already merged.
There is no such commit in the current master, so I think that you force-pushed or whatever, so here is the change once again.

### CODE REVIEW

**Comment by mabar** [comment]
Yeah, I deleted it to release last stable version for nette 2.4. Also it cannot work just like that. We need to load handlers and processors by `Compiler::loadDefinitions()` which don't work in nette 3. If you were using your commit then monolog was probably not logging at all.

https://github.com/contributte/monolog/blob/master/src/DI/MonologExtension.php#L98
https://github.com/nette/di/blob/master/src/DI/Compiler.php#L311

**Comment by solcik** [comment]
I was using it - I registered handlers and processors as services...

https://github.com/contributte/monolog/blob/master/src/DI/MonologExtension.php#L90

**Comment by mabar** [comment]
I see. I forgot about that undocumented behavior. Could you please change `Compiler::loadDefinitions()` to `$compiler->loadDefinitionsFromConfig()`? Ignoring this BC break would broke most of the setups.

**Comment by solcik** [comment]
@mabar updated to DI v3 and schema

**Comment by solcik** [comment]
9abe082 fixes #11

File: src/DI/MonologExtension.php [hunk 1]
```diff
@@ -79,10 +77,6 @@ public function loadConfiguration(): void
 				$channel['handlers']['tracy'] = $tracyHandler;
 			}
 
-			if (!isset($channel['handlers']) || $channel['handlers'] === []) {
-				throw new InvalidStateException(sprintf('%s.channel.%s.handlers must contain at least one handler', $this->name, $name));
```
Comment (mabar) [inline src/DI/MonologExtension.php]: Why did you remove it? There should always be at least one handler

File: composer.json [hunk 2]
```diff
@@ -18,8 +18,9 @@
   "require": {
     "php": ">=7.1.0",
     "monolog/monolog": "^1.23.0",
-    "nette/di": "~2.4.12",
-    "nette/utils": "~2.5.2"
+    "nette/di": "~3.0.0",
+    "nette/utils": "~3.0.0",
+    "nette/schema": "~1.0.0"
```
Comment (f3l1x) [inline composer.json]: nette/schema is part of the nette/di

File: tests/Unit/DI/config_00.neon [hunk 3]
```diff
@@ -0,0 +1 @@
+monolog:
```
Comment (f3l1x) [inline tests/Unit/DI/config_00.neon]: Screenshot 2019-05-30 at 11 55 06

Please move it to `tests/fixtures`

File: composer.json (hunk 2 above)
Comment (mabar) [reply composer.json]: I thinked about the same. And nette/utils also are. Problem is with semantical versioning. nette/di depends on nette/schema^1.0

**Review by solcik — COMMENTED** [review]
I put nette/schema into composer, because it is used in our code. I know, that it is dependency in nette/di, but I am used to require direct dependencies in composer.

File: composer.json (hunk 2 above)
Comment (solcik) [reply composer.json]: I put nette/schema into composer, because it is used in our code. I know, that it is dependency in nette/di, but I am used to require direct dependencies in composer.

Comment (f3l1x) [reply composer.json]: I know guys. But it so tied to nette/di so it's not needed. Thanks for removing it. I will merge this PR after that.
